airflow/providers/jenkins/hooks/jenkins.py

```diff
@@ -46,3 +47,24 @@ def __init__(self, conn_id: str = default_conn_name) -> None:
     def get_jenkins_server(self) -> jenkins.Jenkins:
         """Get jenkins server"""
         return self.jenkins_server
+
+    def get_build_building_state(self, job_name: str, build_number: int) -> bool:
+        """Get build building state"""
+        try:
+            if not build_number:
+                self.log.info(f"Build number not specified, getting latest build info from Jenkins")
+                job_info = self.jenkins_server.get_job_info(self.job_name)
+                build_number_to_check = job_info['lastBuild']['number']
+            else:
+                build_number_to_check = self.build_number
+
+            self.log.info(f"Getting build info for build number: #{build_number_to_check}")
+            build_info = self.jenkins_server.get_build_info(job_name, build_number_to_check)
+            building = build_info['building']
+            return building
+        except jenkins.NotFoundException as err:
+            raise AirflowException(f'Jenkins job status check failed. Final error was: {err.resp.status}')
```

Comment:
According to the source code of `get_job_info` & `get_build_info` there is no case of raising `NotFoundException`
This is an internal exception of the functions but once caught the functions raise `JenkinsException`

https://opendev.org/jjb/python-jenkins/src/commit/d1e469649574883bc531d35fb3aee207f45c15a5/jenkins/__init__.py#L632

https://opendev.org/jjb/python-jenkins/src/commit/d1e469649574883bc531d35fb3aee207f45c15a5/jenkins/__init__.py#L452